server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceAggregationBuilder.java

```diff
@@ -307,30 +285,19 @@ protected final ValuesSourceAggregatorFactory doBuild(QueryShardContext querySha
     }
 
     /**
-     * Provide a hook for aggregations to have finer grained control of the CoreValuesSourceType for script values.  This will only be
-     * called if the user did not supply a type hint for the script.  The script object is provided for reference.
+     * Provide a hook for aggregations to have finer grained control of the {@link ValuesSourceType } when {@link ValuesSourceConfig}
+     * can't resolve a type (i.e.  this will be called before falling back to CoreValuesSourceType.ANY.)
      *
      * @param script - The user supplied script
      * @return The CoreValuesSourceType we expect this script to yield.
      */
-    protected ValuesSourceType resolveScriptAny(Script script) {
+    protected ValuesSourceType defaultValueSourceType(Script script) {
```

Comment:
Bytes seems like a sensible default here, but I'm not convinced this shouldn't just be abstract and force each aggregation builder to specify it.

Reply:
Yeah I was ++ default at first, but after looking through a bunch I'm leaning towards abstract and forcing each agg to implement.  Half of them do anyway (most of the metrics), and it'd be easy to forget to override and might lead to weird unanticipated edge cases?

Reply:
Yeah, I'm coming to that conclusion too.  Kind of want to do it in a follow up PR though, since this is huge already and that'll be a lot of boiler plate.

Reply:
:+1: